build/compile.sh

```diff
@@ -171,22 +171,33 @@ build() {
 
   pushd $PY27
 
-  local compile_readline
-  local link_readline
+  if is_macos; then
+    cp pyconfig.macos.h pyconfig.h
+  else
+    cp pyconfig.linux.h pyconfig.h
+  fi
+
+  local readline_dir
+  local readline_flags
   if test "$HAVE_READLINE" = 1; then
     # Readline interface for tokenizer.c and [raw_]input() in bltinmodule.c.
     # For now, we are using raw_input() for the REPL.  TODO: Parameterize this!
     # We should create a special no_readline_raw_input().
 
     c_module_src_list=$(cat $abs_c_module_srcs)
+
     # NOTE: pyconfig.h has HAVE_LIBREADLINE but doesn't appear to use it?
-    compile_readline='-D HAVE_READLINE'
-    link_readline='-l readline'
+    readline_flags=(-D HAVE_READLINE)
+
+    if is_macos; then
```

Comment:
This logic should use feature detection, not version detection.  See this comment:

https://news.ycombinator.com/item?id=15195370

https://oilshell.zulipchat.com/#narrow/stream/121540-oil-discuss/topic/Shell.20builtins.20and.20toybox

Alternatively, why not just make a homebrew package for OSH and set `--readline-dir` there, and remove this special logic?

I don't think it's safe to assume that all OS X users use HomeBrew.  Someone could have manually installed readline without HomeBrew, and it would live in a different dir.

Reply:
thanks for the advice! feature detection is indeed more robust.

and i think a `--readline-dir` flag is the way to go too